src/log_to_disk.c

```diff
@@ -118,7 +118,13 @@ void log_to_disk(int from_lan) {
 
 	total = total + score2(lan_logline);
 
-	struct qso_t *qso = parse_qso(lan_logline);
+	gchar *tlogline = g_strdup(lan_logline);
+
+	struct qso_t *qso = parse_qso(tlogline);
+
+	if (tlogline != NULL) {
```

Comment:
Check for null can be dropped: _If mem is NULL it simply returns, so there is no need to check mem against NULL before calling this function._
https://docs.gtk.org/glib/func.free.html